### PR TITLE
Tp banner manager fixes

### PR DESCRIPTION
**Pull request by TomPretty**
## What does this change?
A handfull of useability fixes and improvements:
- stop saving tests with erros
- select tests on creation
- select variants on creation
- close delete dialog
- names are now case insensitive
- allow swapping to a new test if nothing has been modified

### CODE REVIEW

File: public/src/components/channelManagement/bannerTests/bannerTestsForm.tsx [hunk 1]
```diff
@@ -57,8 +57,13 @@ const BannerTestsForm: React.FC<Props> = ({
   editMode,
 }: Props) => {
   const createTest = (name: string, nickname: string): void => {
-    const newTests = [...tests, createDefaultBannerTest(name, nickname)];
-    onTestsChange(newTests, name);
+    if (Object.keys(modifiedTests).length > 0) {
```
Comment (paulbrown1982) [inline public/src/components/channelManagement/bannerTests/bannerTestsForm.tsx:60]: I think we should refactor `modifiedTests` to be an array in a future PR.

**Review by paulbrown1982 — APPROVED** [review]
👍  LGTM, and fine to go out to make the UX more reliable for the first user. However I think we should add a card to refactor the `modifiedTests` object to be an array or structured object.

File: public/src/components/channelManagement/bannerTests/bannerTestsForm.tsx (hunk 1 above)
Comment (TomPretty) [reply public/src/components/channelManagement/bannerTests/bannerTestsForm.tsx:60]: Yeah we can actually refactor to get rid of it entirely. It's a hangover from when mutliple tests could be worked on in one go. The new UX is that only one test can be worked on at any one time. Now we can just use a single `modifiedTest` object to keep track of the changes to a single test.

Comment (tomrf1) [reply public/src/components/channelManagement/bannerTests/bannerTestsForm.tsx:60]: 👍 agreed, let's get rid of `modifiedTest` now that things are simpler

**Pull request merged**